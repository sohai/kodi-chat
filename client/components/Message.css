.message {
  width: 70%;
  margin: 10px;
  padding: 10px;
  background-color: #E0E0E0;
  border-radius: 6px;
}
.thinking {
  background-color: #BDBDBD;
}
.incoming {
  align-self: flex-end;
  background-color: #BBDEFB;
}
.incoming.thinking {
  background-color: #448AFF;
}

