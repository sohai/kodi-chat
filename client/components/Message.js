import React from 'react';
import { pure, compose } from 'recompact';
import PropTypes from 'prop-types';
// import ImmutablePropTypes from 'react-immutable-proptypes'
import classNames from 'classnames';
import styles from './Message.css';
import Typography from './Typography';

const propTypes = PropTypes && {};
const defaultProps = {};

export const MessageComponent = ({ item, className: classNameProp }) => {
  const className = classNames(
    styles.message,
    {
      [styles.incoming]: item.type === 'inc',
      [styles.thinking]: item.variant === 'thinking'
    },
    classNameProp
  );
  const textParts = item.message.split(':)');
  return (
    <div className={className}>
      <Typography variant="body">
        {textParts.map((text, idx) => {
          return (
            <span key={idx}>
              {text}
              {idx !== textParts.length - 1 && (
                <img
                  className={styles.emo}
                  src="https://emojipedia-us.s3.amazonaws.com/thumbs/160/twitter/53/slightly-smiling-face_1f642.png"
                />
              )}
            </span>
          );
        })}
      </Typography>
      <Typography fade variant="caption" align="right">
        {item.username}
      </Typography>
    </div>
  );
};

MessageComponent.displayName = 'Message';
MessageComponent.propTypes = propTypes;
MessageComponent.defaultProps = defaultProps;
export default compose(pure)(MessageComponent);
