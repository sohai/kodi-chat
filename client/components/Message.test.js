/* global render, describe, it, expect */
/* eslint-disable import/first */
import React from 'react';
import Message from './Message';

describe('(Component) Message', () => {
  it('should render', () => {
    const props = {
      item: {
        type: 'inc',
        message: 'message',
        thinking: 'thinking'
      }
    };
    const wrapper = render(<Message {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
