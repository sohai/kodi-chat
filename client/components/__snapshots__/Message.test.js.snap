// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Message should render 1`] = `
<div
  class="message incoming"
>
  <div
    class="root dark"
  >
    <span>
      message
    </span>
  </div>
  <div
    class="root dark caption fade alignRight"
  />
</div>
`;
